pageserver/src/layered_repository/interval_tree.rs

```diff
@@ -291,24 +292,20 @@ mod tests {
         if let Some(v) = tree.search(key) {
             let vstr = v.to_string();
 
-            if expected.is_empty() {
-                panic!("search with {} returned {}, expected None", key, v);
-            }
+            assert!(!expected.is_empty(), "search with {} returned {}, expected None", key, v);
```

Comment:
The `!expected.is_empty()` check in conjunction with `expected None` is quite confusing, but I guess I might not know something 🙂

Reply:
It took me a moment to figure out as well. From a higher level, what the function is doing is really just `assert!(result.is_none() == expected.is_empty())` -- it should have returned `None` iff we weren't expecting anything. This is just the case if we already know the result is `Some(_)`.

https://github.com/zenithdb/zenith/blob/d92412554fe81f22c5d3f52173de3a2e82d9f1d1/pageserver/src/layered_repository/interval_tree.rs#L292-L295